bot/cogs/clean.py

```diff
@@ -114,43 +113,37 @@ def predicate_regex(message: Message) -> bool:
         if not channels:
             channels = [ctx.channel]
 
+        # Delete the invocation first
+        with self.mod_log.ignore_all():
+            await ctx.message.delete()
+
         # Look through the history and retrieve message data
+        # This is only done so we can create a log to upload.
         messages = []
         message_ids = []
         self.cleaning = True
-        invocation_deleted = False
 
-        # To account for the invocation message, we index `amount + 1` messages.
         for channel in channels:
-            async for message in channel.history(limit=amount + 1):
+            async for message in channel.history(limit=amount):
```

Comment:
Since you added the `ignore_all` decorator, this loop is obsolete.

Reply:
Looks like channel.purge will return the messages, so you are right - we can just store the messages from the return values here and then use those for the log upload.